File: src/com/dafttech/terra/engine/gui/Tooltip.java
package com.dafttech.terra.engine.gui;

import com.dafttech.terra.engine.Vector2;
import com.dafttech.terra.engine.gui.anchors.AnchorMouse;
import com.dafttech.terra.engine.gui.anchors.GUIAnchorSet;
import com.dafttech.terra.engine.gui.elements.ElementLabel;
import com.dafttech.terra.engine.gui.elements.GUIElement;

public class Tooltip {
    public static ElementLabel label;

    public static void init() {
        label = new ElementLabel(new Vector2(0, 0), "");

        GUIAnchorSet labelSet = new GUIAnchorSet();
        labelSet.addAnchor(new AnchorMouse());

        label.assignAnchorSet(labelSet);
    }

    public static GUIElement getLabel() {
        return label;
    }

    public static void setText(String txt) {
        label.setText(txt);
    }
}

File: src/main/java/com/dafttech/terra/engine/gui/Tooltip.scala
package com.dafttech.terra.engine.gui

import com.dafttech.terra.engine.Vector2
import com.dafttech.terra.engine.gui.anchors.{AnchorMouse, GUIAnchorSet}
import com.dafttech.terra.engine.gui.elements.{ElementLabel, GUIElement}

object Tooltip {
  var label: ElementLabel = null

  def init {
    label = new ElementLabel(new Vector2, "")
    val labelSet: GUIAnchorSet = new GUIAnchorSet
    labelSet.addAnchor(new AnchorMouse)
    label.assignAnchorSet(labelSet)
  }

  def getLabel: GUIElement = {
    return label
  }

  def setText(txt: String) {
    label.setText(txt)
  }
}